rust/lance-index/src/vector/ivf/shuffler.rs

```diff
@@ -276,20 +241,36 @@ impl IvfShuffler {
 
     pub async fn write_unsorted_stream(
         &mut self,
-        data: impl RecordBatchStream + Unpin + 'static,
+        data: impl Stream<Item = Result<RecordBatch>>,
     ) -> Result<()> {
         let object_store = ObjectStore::local();
         let path = self.output_dir.child(UNSORTED_BUFFER);
         let writer = object_store.create(&path).await?;
 
+        let mut data = Box::pin(data.peekable());
+
+        let schema = match data.as_mut().peek().await {
```

Comment:
nit: Let's add a check to verify that `_rowid` and IVF PART ID are in the schema